oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+import org.springframework.security.oauth2.core.OAuth2TokenVerifier;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.resourceserver.web.BearerTokenResolver;
+
+/**
+ * Example configuration:
+ *
+ * oauth2().resourceServer().accessToken()
```

Comment:
I'd simplify the hierarchy some. 

```java
oauth2()
     .resourceServer()
         .jwt()
         .verifiers()
             ...
```

Reply:
Agreed, let's try to simplify the hierarchy. I will add to what @rwinch has proposed:

```
oauth2()
    .resourceServer()
		.accessToken()
		    .verifiers(verifiers)	// verifiers -> List<OAuth2TokenVerifier<T extends AbstractOAuth2Token>>
			.jwt()
			    .jwkSetUri(jwkSetUri)	// URI used to retrieve the JWK Set
			.opaque()
			    .introspectionUri(uri);	// URI for the Introspection Endpoint
            ...
```

I feel like we need that extra hierarchy level in `.accessToken()` to indicate we are configuring a specific Access Token formats.

For the feature (to be implemented) that will allow the user to configure a RSA Public Key we could have something like this:

```
oauth2()
    .resourceServer()
	    .keys()
		    .verifyKey(rsaKey1)		// type java.security.PublicKey - used to verify signature
			.decryptKey(rsaKey2)	// type java.security.PrivateKey - used to decrypt token content
            ...
```

The hierarchy under `.keys()`  can than be leveraged by both `jwt` and `opaque` - no need to define the PK twice. It's a valid use case that the Authorization Server supports both JWT and Opaque (Introspection Endpoint) token formats.